### PR TITLE
Add PersonalSections Frontend Delete Button to Table

### LINKED ISSUE #2: FEATURE: I can delete courses from a personal schedule on the main personal schedule page
### Have you updated the title?

- [X] I have updated the issue's title.

### As a...

user

### I can...

delete courses from a personal schedule on the personal schedule page

### So that...

manipulating personal schedules feels more natural

### Discussion

At the moment, when useres add courses to a personal schedule, the personal schedule can be used to show the entire schedule, but to delete a course, they have to go to an entirely separate page.

For example: 

image

Note that the course cannot be delete here.  To delete, you have to go to this page:

image

Note that deleting a section also deletes the attached lecture, and vice-versa.

What we want to do is add a delete button to this page: 

image

Once that's done, we can probably get rid of the "Courses" menu item and its associated React components completely.

### Acceptance Criteria

- [x] The table shown here will have a column with delete buttons
        image

- [x] When clicking the delete button on that page, the course will be deleted, along with any attached lectures or sections.

### Implementation Todos

Frontend:
- [x] Locate the react component that implements that table on the url <https://courses-qa.dokku-00.cs.ucsb.edu/personalschedules/details/4>
- [x] If it does not yet have a storybook entry, create one
- [x] Locate the react component that implements the route <https://courses-qa.dokku-00.cs.ucsb.edu/courses/list>
- [x] In the component that implements the route <https://courses-qa.dokku-00.cs.ucsb.edu/courses/list> there will be code for a delete column.  Use that code as a basis to add a delete column to the other react component
- [x] Test it in both the storybook and in the working app
- [x] Make sure that for classes with connected lecture/section that deleting one automatically deletes the other.

Backend:
* There are likely no backend changes needed

### PR DESCRIPTION
**Pull request by WinbertZhang**
Merge Order:
1. #28 
2. #29 
3. #24 

This PR adds a delete button to the personal sections table allowing users to remove all associated lectures and sections for a course. 

DOKKU Deployment: https://team04-winbertzhang-dev02.dokku-04.cs.ucsb.edu/

Steps:
Login, then create a Personal Schedule at https://team04-winbertzhang-dev02.dokku-04.cs.ucsb.edu/personalschedules/list

Add a course to the personal schedule at https://team04-winbertzhang-dev02.dokku-04.cs.ucsb.edu/

Navigate to the personal schedule details page at https://team04-winbertzhang-dev02.dokku-04.cs.ucsb.edu/personalschedules/details/1

Screenshot 2024-05-25 at 7 11 22 PM

Press on the delete button for any course, and the result is as follows:
Screenshot 2024-05-25 at 7 11 44 PM

Closes #2

### CODE REVIEW

**Review by tihdiR — APPROVED** [review]
LGTM

**Review by AnikaArora — CHANGES_REQUESTED** [review]
Please include the dokku link

**Comment by WinbertZhang** [comment]
> Please include the dokku link

Updated PR Description with dokku link: https://team04-winbertzhang-dev02.dokku-04.cs.ucsb.edu/

**Comment by WinbertZhang** [comment]
This PR is to be merged after #28 is merged.

File: frontend/src/main/utils/sectionUtils.js [hunk 1]
```diff
@@ -104,3 +105,20 @@ export const renderInfoLink = ({ cell: { value } }) => (
...
+
+export function onDeleteSuccess(message) {
+  console.log(message);
+  toast(message);
+}
+
+export function cellToAxiosParamsDelete({ cell, psId }) {
+  const enrollCode = cell.row.values["classSections[0].enrollCode"];
+  return {
+    url: "/api/personalSections/delete",
```
Comment (pconrad) [inline frontend/src/main/utils/sectionUtils.js:117]:
```suggestion
    url: "/api/courses/user",
```

Why not use the existing endpoint?

File: frontend/src/main/utils/sectionUtils.js [hunk 2]
```diff
@@ -104,3 +105,20 @@ export const renderInfoLink = ({ cell: { value } }) => (
...
+    method: "DELETE",
+    params: {
+      psId: psId,
+      enrollCd: enrollCode,
```
Comment (pconrad) [inline frontend/src/main/utils/sectionUtils.js:121]:
```suggestion
     id: id,
```

File: frontend/src/main/utils/sectionUtils.js [hunk 3]
```diff
@@ -104,3 +105,20 @@ export const renderInfoLink = ({ cell: { value } }) => (
...
   </p>
 );
+
+export function onDeleteSuccess(message) {
+  console.log(message);
+  toast(message);
+}
+
+export function cellToAxiosParamsDelete({ cell, psId }) {
+  const enrollCode = cell.row.values["classSections[0].enrollCode"];
```
Comment (pconrad) [inline frontend/src/main/utils/sectionUtils.js:115]:
```suggestion

```

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**